### PR TITLE
[AIRFLOW-5586] Improve CLI error messaging

### PR DESCRIPTION
**Pull request by leahecole**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5586
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Review by mik-laj — APPROVED** [review]
Screenshot 2019-10-02 at 23 56 56
Thanks a lot. I also have a problem with this sometime problem.

**Comment by mik-laj** [comment]
This is a change in the code. Can you create a Jira ticket?

**Comment by leahecole** [comment]
I'm trying this in travis on my own fork but am having trouble recreating the failing part of the travis job locally. Any insight is appreciated!

File: airflow/bin/cli.py [hunk 1]
```diff
@@ -1376,8 +1376,10 @@ def users_list(args):
 def users_create(args):
     appbuilder = cached_appbuilder()
     role = appbuilder.sm.find_role(args.role)
+    valid_roles = appbuilder.sm.get_all_roles()
+    print(valid_roles)
```
Comment (Fokko) [inline airflow/bin/cli.py]: Can we remove the print?

File: airflow/bin/cli.py [hunk 2]
```diff
@@ -1376,8 +1376,10 @@ def users_list(args):
 def users_create(args):
     appbuilder = cached_appbuilder()
     role = appbuilder.sm.find_role(args.role)
+    valid_roles = appbuilder.sm.get_all_roles()
```
Comment (Fokko) [inline airflow/bin/cli.py]: Can you move this inside `if` statement where it is used?

File: airflow/bin/cli.py [hunk 3]
```diff
@@ -1433,8 +1435,9 @@ def users_manage_role(args, remove=False):
             args.username or args.email))
 
     role = appbuilder.sm.find_role(args.role)
+    valid_roles = appbuilder.sm.get_all_roles()
```
Comment (Fokko) [inline airflow/bin/cli.py]: Can you move this inside of the `if` statement?

File: airflow/bin/cli.py [hunk 4]
```diff
@@ -1516,8 +1519,9 @@ def _import_users(users_list):
         roles = []
         for rolename in user['roles']:
             role = appbuilder.sm.find_role(rolename)
+            valid_roles = appbuilder.sm.get_all_roles()
```
Comment (Fokko) [inline airflow/bin/cli.py]: Can you move this inside of the `if` statement?

**Review by Fokko — COMMENTED** [review]
Thank you for the PR. I have a few suggestions. PTAL

File: airflow/bin/cli.py (hunk 1 above)
Comment (leahecole) [reply airflow/bin/cli.py]: Definitely

File: airflow/bin/cli.py (hunk 2 above)
Comment (leahecole) [reply airflow/bin/cli.py]: Done

File: airflow/bin/cli.py (hunk 4 above)
Comment (leahecole) [reply airflow/bin/cli.py]: Done

File: airflow/bin/cli.py (hunk 3 above)
Comment (leahecole) [reply airflow/bin/cli.py]: Done

**Comment by leahecole** [comment]
Great suggestion @Fokko  - thanks!

**Review by Fokko — APPROVED** [review]

**Comment by Fokko** [comment]
My pleasure @leahecole, thank you for the contribution.

**Pull request merged**